src/tls/boring.rs

```diff
@@ -238,6 +240,37 @@ pub fn grpc_connector(uri: String, root_cert: RootCert) -> Result<TlsGrpcChannel
     Ok(TlsGrpcChannel { uri, client: hyper })
 }
 
+// creating K8s client with boring
+pub async fn create_k8s_client(root_cert: RootCert) -> Result<kube::Client, anyhow::Error> {
+    let mut conn = ssl::SslConnector::builder(ssl::SslMethod::tls_client())?;
+    conn.set_verify(ssl::SslVerifyMode::PEER);
+    conn.set_alpn_protos(Alpn::H2.encode())?;
```

Comment:
Is this the only SSL connection we create in ztunnel ? Any chance we could use the same code - or client library code ?